ddl/placement_policy.go

```diff
@@ -29,21 +29,47 @@ import (
 
 func onCreatePlacementPolicy(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	policyInfo := &model.PolicyInfo{}
-	if err := job.DecodeArgs(policyInfo); err != nil {
+	var orReplace bool
+	if err := job.DecodeArgs(policyInfo, &orReplace); err != nil {
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
 	policyInfo.State = model.StateNone
 
-	err := checkPlacementPolicyNotExistAndCancelExistJob(d, t, job, policyInfo)
-	if err != nil {
+	if err := checkPolicyValidation(policyInfo.PlacementSettings); err != nil {
+		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
-	err = checkPolicyValidation(policyInfo.PlacementSettings)
+
+	existPolicy, err := getPlacementPolicyByName(d, t, policyInfo.Name)
 	if err != nil {
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
+
+	if existPolicy != nil {
+		if !orReplace {
```

Comment:
is there any chance OnExistIgnore policy ddl reach here 🤔？

Reply:
Good point, it is possible when a placement policy(say `S`) does not exist before a DDL job is submitted( https://github.com/pingcap/tidb/pull/32986/files#diff-b7c54f7dad7257cbe4fa04b8d76c9bed694e2e2e7c5599f45eb2172d8d4ff8a7R2329) but being created when the DDL job is in the queue and executed here.

It happens when tow `CREATE PLACEMENT POLICY IF NOT EXISTS ...` executed concurrently in multiple clients. However, reporting an error might be a better result for the users, although it seems weird.